opta/core/cloud_provider.py

```diff
@@ -0,0 +1,52 @@
+import boto3
+from botocore.config import Config
+from botocore.exceptions import ClientError
+
+from opta.core.terraform2 import TerraformFile
+from opta.layer2 import Layer
+
+
+class CloudProvider:
+    def state_storage(self, layer: Layer) -> str:
+        return f"opta-tf-state-{layer.org_name}-{layer.name}"
+
+    def verify_storage(self, layer: Layer) -> bool:
+        raise NotImplementedError()
+
+    def add_provider_config(self, file: TerraformFile, layer: Layer) -> None:
```

Comment:
is there a use case to call this method more than once?
we could rename this to `configure_providers` to reflect this